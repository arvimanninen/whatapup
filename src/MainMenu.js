import React from 'react';
import './mainmenu.css';

class MainMenu extends React.Component {
    render() {
        const categoryButtons = this.props.categories.map((category) => 
            <EquipmentCategoryButton key={category.key} category={category} 
            handleViewChange={this.props.handleViewChange}>
            </EquipmentCategoryButton>
        );
        return(
            <div className="container-fluid">
                <div className="row">
                    <div className="col-sm-1 col-md-2 col-lg-3"></div>
                    <div className="col text-center">
                        <header>
                            <h1>WhataPup</h1>
                            <h2>To-do app for the dog puppy acquirers </h2>
                        </header>
                        <div>
                            {categoryButtons}
                        </div>
                        <footer>
                            <a href="https://www.kennelliitto.fi/en/dogs/dont-support-puppy-mills">Don't support puppy mills!</a>
                            <br/>
                            Information based on “Iloinen Koira” (The Happy Puppy)-brochure (Finnish Kennel Club *YEAR*)
                            <br/>
                            App made by Arvi Manninen 2018
                            <br/>
                            Released under MIT License
                        </footer>
                    </div>
                    <div className="col-sm-1 col-md-2 col-lg-3"></div>
                </div>
            </div>
        );
    }
}

class EquipmentCategoryButton extends React.Component {
    render() {
        const categoryName = this.props.category.name;
        return(
            <button className="btn btn-primary btn-lg" onClick={() => this.props.handleViewChange("EquipmentList", categoryName)}>{categoryName}</button>
        );
    }
}

export default MainMenu;