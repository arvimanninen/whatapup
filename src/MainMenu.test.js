import React from 'react';
import renderer from 'react-test-renderer';
import MainMenu from './MainMenu';
import CATEGORIES from './index';

test('Renders MainMenu', () => {
  const component = renderer.create(<MainMenu categories={CATEGORIES}></MainMenu>,);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});