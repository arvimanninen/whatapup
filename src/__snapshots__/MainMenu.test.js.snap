// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders MainMenu 1`] = `
<div>
  <header>
    <h1>
      WhataPup
    </h1>
    <h2>
      To-do app for the dog puppy acquirers 
    </h2>
  </header>
  <div>
    <button
      onClick={[Function]}
    >
      Category A
    </button>
    <button
      onClick={[Function]}
    >
      Category B
    </button>
    <button
      onClick={[Function]}
    >
      Category C
    </button>
  </div>
  <footer>
    <a
      href="https://www.kennelliitto.fi/en/dogs/dont-support-puppy-mills"
    >
      Don't support puppy mills!
    </a>
    <p>
      Information based on “Iloinen Koira” (The Happy Puppy)-brochure (Finnish Kennel Club *YEAR*)
      <br />
      App made by Arvi Manninen 2018
      <br />
      Released under MIT License
      <br />
    </p>
  </footer>
</div>
`;
